src/websocket/index.ts

```diff
@@ -297,3 +322,27 @@ const constructRPCErrorRes = (
     },
   }
 }
+
+const cleanupIntervalMs = CONFIG.websocket.cleanupIntervalMs; // Run cleanup every 10 minutes
+
+const cleanupStaleConnections = () => {
+  connectionsByIP.forEach((sockets, ip) => {
+    sockets.forEach((socket) => {
```

Comment:
this code may have some issue because of concurrent modification. please check once and fix if need be.

Reply:
handled